src/main/drivers/accgyro/accgyro_mpu.c

```diff
@@ -177,37 +219,128 @@ bool mpuGyroRead(gyroDev_t *gyro)
     return true;
 }
 
+
 #ifdef USE_SPI_GYRO
 bool mpuAccReadSPI(accDev_t *acc)
 {
-    STATIC_DMA_DATA_AUTO uint8_t dataToSend[7] = {MPU_RA_ACCEL_XOUT_H | 0x80, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF};
-    STATIC_DMA_DATA_AUTO uint8_t data[7];
+    switch (acc->gyro->gyroModeSPI) {
+    case GYRO_EXTI_INT:
+    case GYRO_EXTI_NO_INT:
+    {
+        // Ensure any prior DMA has completed before continuing
+        spiWaitClaim(&acc->gyro->dev);
 
-    const bool ack = spiReadWriteBufRB(&acc->gyro->dev, dataToSend, data, 7);
-    if (!ack) {
-        return false;
+        acc->gyro->dev.txBuf[0] = MPU_RA_ACCEL_XOUT_H | 0x80;
+
+        busSegment_t segments[] = {
+                {NULL, NULL, 7, true, NULL},
+                {NULL, NULL, 0, true, NULL},
+        };
+        segments[0].txData = acc->gyro->dev.txBuf;
+        segments[0].rxData = &acc->gyro->dev.rxBuf[1];
+
+        spiSequence(&acc->gyro->dev, &segments[0]);
+
+        // Wait for completion
+        spiWait(&acc->gyro->dev);
+
+        // Fall through
+        FALLTHROUGH;
     }
 
-    acc->ADCRaw[X] = (int16_t)((data[1] << 8) | data[2]);
-    acc->ADCRaw[Y] = (int16_t)((data[3] << 8) | data[4]);
-    acc->ADCRaw[Z] = (int16_t)((data[5] << 8) | data[6]);
+    case GYRO_EXTI_INT_DMA:
+    {
+        // If read was triggered in interrupt don't bother waiting. The worst that could happen is that we pick
+        // up an old value.
+
+        // This data was read from the gyro, which is the same SPI device as the acc
+        uint16_t *accData = (uint16_t *)acc->gyro->dev.rxBuf;
+        acc->ADCRaw[X] = __builtin_bswap16(accData[1]);
+        acc->ADCRaw[Y] = __builtin_bswap16(accData[2]);
+        acc->ADCRaw[Z] = __builtin_bswap16(accData[3]);
+        break;
+    }
+
+    case GYRO_EXTI_INIT:
+    default:
+        break;
+    }
 
     return true;
 }
 
 bool mpuGyroReadSPI(gyroDev_t *gyro)
 {
-    STATIC_DMA_DATA_AUTO uint8_t dataToSend[7] = {MPU_RA_GYRO_XOUT_H | 0x80, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF};
-    STATIC_DMA_DATA_AUTO uint8_t data[7];
+    uint16_t *gyroData = (uint16_t *)gyro->dev.rxBuf;
+    switch (gyro->gyroModeSPI) {
+    case GYRO_EXTI_INIT:
+    {
+        // Initialise the tx buffer to all 0xff
+        memset(gyro->dev.txBuf, 0xff, 16);
+#ifdef USE_GYRO_EXTI
+        // Check that minimum number of interrupts have been detected
+
+        // We need some offset from the gyro interrupts to ensure sampling after the interrupt
+        gyro->gyroDmaMaxDuration = 5;
+        // Using DMA for gyro access upsets the scheduler on the F4
+        if (gyro->detectedEXTI > GYRO_EXTI_DETECT_THRESHOLD) {
+            if (spiUseDMA(&gyro->dev)) {
```

Comment:
can you rename `spiUseDMA` to `spiDMAIsUsed` as it reads like it's telling the SPI system to use DMA, when in-fact it's checking to see if it is used.